### PR TITLE
Add NOI Subtypes to Admin

### PR DESCRIPTION
**Pull request by dhaselhan**
* Add active flag to enable/disable
* Add Admin controller and services to back
* Add page to admin

### CODE REVIEW

File: alcs-frontend/src/app/features/admin/noi-subtype/noi-subtype.component.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          <mat-icon color="warn">not_interested</mat-icon>
+          Inactive
+        </div>
+      </td>
+    </ng-container>
+
+    <ng-container matColumnDef="actions">
+      <th mat-header-cell *matHeaderCellDef>Actions</th>
+      <td mat-cell *matCellDef="let row">
+        <button *ngIf="row.code !== 'MODI'" class="edit-btn" mat-flat-button (click)="onEdit(row)">
```
Comment (mhuseinov) [inline alcs-frontend/src/app/features/admin/noi-subtype/noi-subtype.component.html:48]: MODI? is it from modification or it is a subtype type?

File: alcs-frontend/src/app/services/noi-subtype/noi-subtype.service.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      console.log(e);
+    }
+    return;
+  }
+
+  async delete(code: string) {
+    try {
+      return await firstValueFrom(this.http.delete<CeoCriterionDto>(`${this.url}/${code}`));
+    } catch (e) {
+      this.toastService.showErrorToast('Failed to delete ceo criterion');
```
Comment (mhuseinov) [inline alcs-frontend/src/app/services/noi-subtype/noi-subtype.service.ts:51]: typo in message: Failed to delete NOI subtype

File: alcs-frontend/src/app/services/notice-of-intent/notice-of-intent.service.ts [hunk 3]
```diff
@@ -73,4 +73,8 @@ export class NoticeOfIntentService {
       return undefined;
     }
   }
+
+  async deleteSubtype(code: string) {
+    //TODO
```
Comment (mhuseinov) [inline alcs-frontend/src/app/services/notice-of-intent/notice-of-intent.service.ts:78]: do we need this?

**Review by mhuseinov — APPROVED** [review]

File: alcs-frontend/src/app/services/notice-of-intent/notice-of-intent.service.ts (hunk 3 above)
Comment (dhaselhan) [reply alcs-frontend/src/app/services/notice-of-intent/notice-of-intent.service.ts:78]: Removed

File: alcs-frontend/src/app/services/noi-subtype/noi-subtype.service.ts (hunk 2 above)
Comment (dhaselhan) [reply alcs-frontend/src/app/services/noi-subtype/noi-subtype.service.ts:51]: Fixed

File: alcs-frontend/src/app/features/admin/noi-subtype/noi-subtype.component.html (hunk 1 above)
Comment (dhaselhan) [reply alcs-frontend/src/app/features/admin/noi-subtype/noi-subtype.component.html:48]: Removed

**Pull request merged**